### ISSUE TITLE
IOS Connect on Demand

### ISSUE THREAD

**Issue by InventorB** [netbirdio/netbird#2214]
**Is your feature request related to a problem? Please describe.**
I've been using your software to connect into my home network thousands of miles away. However, I consistently keep finding the VPN disconnecting. It happens around multiple times a day and is a barrier to me liking the software when it keeps disconnecting. 

**Describe the solution you'd like**
Apparently IOS has an API called Connect on Demand where it will auto-reconnect every time the network changes with NO USER INTERACTION REQUIRED. Practically every other major VPN app supports it, and this feature is very effective. When properly done, it should even auto-connect after a restart.

Here's Apple's documentation page on the feature. This does not look hard to implement.
https://developer.apple.com/documentation/networkextension/personal_vpn/vpn_on_demand_rules

It would be more amazing if you could even create a GUI to exclude networks and interfaces (like disable on the Home WiFi SSID, enable only on cellular, only enable on Ethernet), but I understand that's time and effort the developers may not have. But if you can add this (now or future), I would love it and I'm sure many others would also.
**If that does sound like too much for now just enable it on all interfaces.**

**Describe alternatives you've considered*
I tried IOS shortcuts, and while that can help, it's too much effort for the many users and Shortcuts is an inefficient method that does not solve the problem.

**Thanks for considering this improvement.**

**Comment by ramphex** [comment]
also want this

**Comment by fairct** [comment]
I agree that this would be a great feature to have - but I think this is a duplicate of https://github.com/netbirdio/netbird/issues/2064
